tests/TestCase/BasicsTest.php

```diff
@@ -40,19 +40,19 @@ public function testArrayDiffKey()
         $two = ['one' => 'one', 'two' => 'two'];
         $result = array_diff_key($one, $two);
         $expected = ['three' => 3];
-        $this->assertEquals($expected, $result);
+        $this->assertSame($expected, $result);
 
         $one = ['one' => ['value', 'value-two'], 'two' => 2, 'three' => 3];
         $two = ['two' => 'two'];
         $result = array_diff_key($one, $two);
         $expected = ['one' => ['value', 'value-two'], 'three' => 3];
-        $this->assertEquals($expected, $result);
+        $this->assertSame($expected, $result);
```

Comment:
Won't assertSame() fail if the array key order changes? Is that a useful constraint to put on ourselves?

Reply:
Yeah assertSame() should be used for scalars only not arrays.

Reply:
In simple arrays like this it might also OK, but I see your point.

Reply:
So should we keep these simple arrays in strict assert? It has the benefit of checking all the types here correctly, too.
As long as their order is to be expected exactly like this, we can keep them IMO to have the better assert. More high level comparisons sure should stay less strict.